lib/elixir/src/elixir_fn.erl

```diff
@@ -129,7 +129,7 @@ validate(_Meta, [], _Pos, _E) ->
   [].
 
 escape({'&', _, [Pos]}, _E, Dict) when is_integer(Pos), Pos > 0 ->
-  Var = {list_to_atom([$x | integer_to_list(Pos)]), [], ?var_context},
+  Var = {list_to_atom([$& | integer_to_list(Pos)]), [], nil},
```

Comment:
Let's drop a comment why the context is nil. This can end-up poluting the user space (but the likelihook is very low).